src/main/scala/mesosphere/marathon/api/akkahttp/SystemController.scala

```diff
@@ -20,23 +23,40 @@ import scala.concurrent.ExecutionContext
 /**
   * The SystemController handles system level functionality like configuration, metrics and logging.
   */
-class SystemController(cfg: Config)(
+class SystemController(val marathonConfig: MarathonConf, val cfg: Config, val electionService: ElectionService)(
     implicit
     val actorSystem: ActorSystem,
     val executionContext: ExecutionContext,
     val authenticator: Authenticator,
-    val authorizer: Authorizer,
-    val electionService: ElectionService
-) extends Controller with StrictLogging {
+    val authorizer: Authorizer
+) extends Controller with MetricsConversion with StrictLogging {
 
   import Directives._
   import EntityMarshallers._
 
+  def acceptsTextPlain(ranges: Seq[MediaRange]): Boolean = {
+    ranges.exists(range => range.matches(MediaTypes.`text/plain`))
+  }
+  def acceptsApplicationJson(ranges: Seq[MediaRange]): Boolean = {
+    ranges.exists(range => range.matches(MediaTypes.`application/json`))
+  }
   /**
     * GET /ping
     * @return a simple pong as text/plain
     */
-  def ping: Route = complete("pong")
+  def ping: Route = extract(_.request.header[Accept]) { maybeAccept: Option[Accept] =>
+    // Match by priority. Always prefer text/plain.
+    maybeAccept match {
```

Comment:
Akka http supports content negotiation :) for example: 
```scala
val respFuture = Marshal(response).toResponseFor(request) // with content negotiation!
```

Reply:
Wait, it actually should work implicitly, you just need some marshallers

Reply:
Hm. I wonder if we really to go the implicit route here. I don't a benefit in hiding it just for this request. If we'd support this behaviour for more than one route, I'd agree with you. I'll look into to the content negotiation.